src/main/java/me/totalfreedom/totalfreedommod/util/History.java

```diff
@@ -0,0 +1,98 @@
+package me.totalfreedom.totalfreedommod.util;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.UUID;
+import org.bukkit.scheduler.BukkitRunnable;
+import org.bukkit.ChatColor;
+import org.bukkit.command.CommandSender;
+import me.totalfreedom.totalfreedommod.TotalFreedomMod;
+
+public class History
+{
+    public static void reportHistory(final CommandSender sender, final String username)
+    {
+        new BukkitRunnable() 
+        {
+            @Override
+            public void run()
+            {
+                UUID uuid;
+                FHistory history = null;
+                uuid = UUIDFetcher.fetch(username);
+                if (uuid != null)
+                {
+                    Gson gson = new GsonBuilder().create();
+                    String compactUuid = uuid.toString().replace("-", "");
+                    try
+                    {
+                        URL url = new URL("https://api.mojang.com/user/profiles/" + compactUuid + "/names");
+                        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+                        BufferedReader reader = new BufferedReader(new InputStreamReader(conn.getInputStream()));  
+                        FName[] oldNames = gson.fromJson(reader, FName[].class);
+                        reader.close();
+                        conn.disconnect();
+                        Arrays.sort(oldNames);
+                        history = new FHistory(uuid, oldNames);
+                        printHistory(sender, oldNames);
```

Comment:
As the Bukkit API is not thread safe, you cannot call this method asynchronously. You have to resynchronise with the server first.